### PR TITLE
Allow building of rides and placing of staff in Scenario Editor's Landscape Editor mode.

### PR DESCRIPTION
**Pull request by eezstreet**
Pretty much does as you would expect it to do. It allows you to hire staff and build rides in the Scenario Editor while in the Landscape Editor mode. 

No netcode changes should be required.

### CODE REVIEW

File: distribution/changelog.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@
 - Feature: [objects#198] Add additional pirate roofs.
 - Feature: [objects#205] Add additional glass roofs.
 - Feature: [objects#209] Add the Steel Roller Coaster train and 2-across Inverted Train from RollerCoaster Tycoon 1.
+- Feature: [#00000] Allow building of rides in Landscape Editor mode in Scenario Editor
```
Comment (SpartanFrederic104) [inline distribution/changelog.txt:18]: 18150?

**Comment by karst** [comment]
Sandbox Mode should be enabled so people can build rides outside the park perimeters. 

I am not 100% about this one, but I do think it's also smart to make the rides ignore research status.

Staff is frozen solid (flying in the air when placing them down).

Rides can break down while building up the park (and due to previous issue will not be repaired).

**Comment by ocalhoun6** [comment]
> Sandbox Mode should be enabled so people can build rides outside the park perimeters.

Could cause issues, especially with peep pathfinding as they try to navigate to rides outside the park, then get frustrated because they can't find it.

Sandbox mode should probably be *available*, but not enabled by default.

>but I do think it's also smart to make the rides ignore research status.

Definitely. When building a scenario, you can place scenery that hasn't been researched yet, right?

>Staff is frozen solid (flying in the air when placing them down).

Yes, ideally staff should fall down onto the path an *then* freeze. That way you don't get the odd sight of all staff simultaneously falling to the ground at the moment you start the scenario. I do like that they freeze in place, though. That helps if the scenario builder wants the staff to be in specific places when the scenario starts. If they were allowed to wander all over the place as you build the scenario, details like that could be very hard to manage. 

Also, if you're not finished with the scenario's path system yet, having staff wandering around might result in some staff stuck on isolated paths no longer connected to the main system, or some staff ending up not on a path at all.

>Rides can break down while building up the park (and due to previous issue will not be repaired).

Yes, definitely. All ride breakdowns should be disabled while in scenario building mode.

**Comment by karst** [comment]
> Could cause issues, especially with peep pathfinding as they try to navigate to rides outside the park, then get frustrated because they can't find it. Also with mechanics trying to get to those rides and inspect them. 

It's already possible when enabling sandbox mode while playing the game, I don't see why not here.

> Sandbox mode should probably be available, but not enabled by default.

It is already enabled, since you're building a scenario. You can build everything outside the perimeters already, except for rides in current form.

**Comment by Gymnasiast** [comment]
This seems _very_ susceptible to bugs to me. I’m also not 100% sure how useful it is considering we have made the Scenario Editor tools available in regular gameplay.

**Comment by ocalhoun6** [comment]
But things like the terrain generator aren't available (yet) in regular gameplay, right?

I guess, ultimately, the goal would be to merge all the scenario editor functions into regular gameplay ... and then, instead of the scenario editor being its own separate mode, it would simply be launching a blank scenario + opening the first of the scenario creation wizard windows. (It would still look much the same as it does now, but under the hood, it would mostly be using the code for normal gameplay.)

**Comment by Gymnasiast** [comment]
Discussed this a bit in the team chat, and our plan is indeed more along the lines of what @ocalhoun6 describes.

**Pull request closed without merging**